### PR TITLE
Add more debug information to swapchain errors in Vulkan context

### PR DESCRIPTION
**Pull request by clayjohn**
Part of https://github.com/godotengine/godot/issues/71929

We are getting a lot of random lockups that either spam or print:
```
E 0:00:01:0015   VulkanContext::swap_buffers: Condition "err" is true. Returning: ERR_CANT_CREATE
  <C++ Source>   drivers\vulkan\vulkan_context.cpp:2299 @ VulkanContext::swap_buffers()
E 0:00:01:0016   VulkanContext::prepare_buffers: Vulkan: Did not create swapchain successfully.
  <C++ Error>    Condition "err != VK_SUCCESS" is true. Breaking.
  <C++ Source>   drivers\vulkan\vulkan_context.cpp:2222 @ VulkanContext::prepare_buffers()
```

Unfortunately, the lockup happens randomly and it is very difficult to reproduce. 

This PR adds the actual VKResult error to the print so user reports can tell us what Vulkan error is being produced and distinguishes the print statements so we can identify where they are coming from better.

### CODE REVIEW

File: drivers/vulkan/vulkan_context.cpp [hunk 1]
```diff
@@ -2348,7 +2348,10 @@ Error VulkanContext::swap_buffers() {
 	submit_info.signalSemaphoreCount = 1;
 	submit_info.pSignalSemaphores = &draw_complete_semaphores[frame_index];
 	err = vkQueueSubmit(graphics_queue, 1, &submit_info, fences[frame_index]);
-	ERR_FAIL_COND_V(err, ERR_CANT_CREATE);
+
+	if (err != VK_SUCCESS) {
```
Comment (akien-mga) [inline drivers/vulkan/vulkan_context.cpp:2352]: The if is redundant, the condition should go in the `ERR_FAIL_COND_V_MSG`. Same below.

**Review by RandomShaper — APPROVED** [review]

File: drivers/vulkan/vulkan_context.cpp [hunk 2]
```diff
@@ -2348,7 +2348,10 @@ Error VulkanContext::swap_buffers() {
...
+		ERR_FAIL_COND_V_MSG(err, ERR_CANT_CREATE, "Err: " + String(string_VkResult(err)));
```
Comment (akien-mga) [inline drivers/vulkan/vulkan_context.cpp:2353]: I think we can make the error message a bit more explicit. This is fine for us debugging something we can reproduce, but here these are errors that users will see, try to understand, and eventually share with us. It's best if it's written to be somewhat explicit.

File: drivers/vulkan/vulkan_context.cpp [hunk 3]
```diff
@@ -2271,7 +2271,7 @@ Error VulkanContext::prepare_buffers() {
 				print_verbose("Vulkan: Early suboptimal swapchain.");
 				break;
 			} else if (err != VK_SUCCESS) {
-				ERR_BREAK_MSG(err != VK_SUCCESS, "Vulkan: Did not create swapchain successfully.");
+				ERR_BREAK_MSG(err != VK_SUCCESS, "Vulkan: Did not create swapchain successfully." + String(string_VkResult(err)));
```
Comment (akien-mga) [inline drivers/vulkan/vulkan_context.cpp:2274]: An error enum that comes after a dot seems a bit weird. I'd prefix with "Error code: " or similar.

**Comment by clayjohn** [comment]
@akien-mga Thank you for the comments. I have addressed them all and force pushed

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**